clients/src/main/java/org/apache/kafka/common/requests/ListOffsetRequest.java

```diff
@@ -88,7 +87,7 @@ public Builder setTargetTimes(Map<TopicPartition, Long> partitionTimestamps) {
         public ListOffsetRequest build() {
             short version = version();
             if (version < minVersion) {
-                throw new ObsoleteBrokerException("The broker is too old to send this request.");
+                throw new UnsupportedVersionException("The broker is too old to understand this request.");
```

Comment:
I feel bad for the broker. It's not too old; it's just fallen behind the times a bit. 

By the way, could we add some more info to this message, such as which api is used, which version is needed and which is supported by the broker. Same for the other below.